data/transactions/verify/txn.go

```diff
@@ -59,24 +59,29 @@ const txnPerWorksetThreshold = 32
 // - it allows us to linearly scan the input, and process elements only once we're going to queue them into the pool.
 const concurrentWorksets = 16
 
-// GroupContext is the set of parameters external to a transaction which
-// stateless checks are performed against.
+// GroupContext holds values used to evaluate the LogicSigs in a group.  The
+// first set are the parameters external to a transaction which could
+// potentially change the result of LogicSig evaluation. Example: If the
+// consensusVersion changes, a rule might change that matters. Certainly this is
+// _very_ rare, but we don't want to use the result of a LogicSig evaluation
+// across a protocol upgrade boundary.
 //
-// For efficient caching, these parameters should either be constant
-// or change slowly over time.
-//
-// Group data are omitted because they are committed to in the
-// transaction and its ID.
+// The second set are derived from the first set and from the transaction
+// data. They are stored here only for efficiency, not for correctness, so they
+// are not checked in Equal()
 type GroupContext struct {
+	// These fields determine whether a logicsig must be re-evaluated.
```

Comment:
Notice that I've removed minAvmVersion.  Please convince yourself that was correct. My claim is that minAvmVersion can only change for a group is the consensusVersion changes (we decide to change the rules). So there's no need to explicitly track minAvmVersion too.

Reply:
It seems like the min avm version only changes if there's a rekey or an appcall in the group.  Any other requirement on the minimum version would have to come from a consensus change.

Reply:
So basically put all of this:
```
	version, pc, err := versionCheck(program, params.Proto)
	if err != nil {
		return err
	}

	var cx EvalContext
	cx.version = version
	cx.pc = pc
...
...
	if err := cx.minAvmVersionCheck(); err != nil {
		return err
	}
```
into a cx method?

I like that, assuming it also works for `check()`, not just `eval()`. Let me look into it.